lib/wrapeth/provider.ex

```diff
@@ -10,6 +10,14 @@ defmodule Wrapeth.Provider do
 
       @node_url Application.compile_env(@otp_app, __MODULE__)[:node_url]
 
+      unless @client_type do
+        raise RuntimeError, "Client type not defined in configuration"
+      end
+
+      unless @node_url do
+        raise RuntimeError, "Node URL not defined in configuration"
+      end
```

Comment:
This is fine, but not exactly what I meant.
My previous comments were directed to the example you have.
I was hoping to see something like:
```elixir
node_url: System.get_env("ETH_NODE_URL") || raise "Error message here"
```

Reply:
I considered your suggestion, and I think that implementing the fallback directly in the code, rather than instructing the user to define it, would be more user-friendly. That's why I made the modification directly in the code. What do you think?

Reply:
Given that we are checking for this variable's existence at compile time, then there's no harm in having those checks in the provider. But if we start checking environment variable values at runtime, this code would probably need to change.
Another thing: Given these are compile-time checks, is it ok to raise `RuntimeError`s?

Reply:
In fact, it does not make sense. I just had a look at the documentation, and the only compilation exception mentioned is 'CompileError.' However, I believe it would be better to just raise 'Client type not defined in configuration.' Is that correct, or should I raise the 'CompileError' exception?

Reply:
What you did is just fine 👌